packages/components/src/Divider.astro

```diff
@@ -0,0 +1,46 @@
+---
+import { classMerge } from "@gemini-ui-astro/core";
+
+export interface DividerProps {
+  class?: string;
+  thickness?: string;
+  orientation?: "horizontal" | "vertical";
+  color?: string;
+  mt?: string;
+  mb?: string;
+  ml?: string;
+  mr?: string;
+}
+
+const {
+  class: className,
+  thickness = "1px",
+  orientation = "horizontal",
+  color = "#323232",
+  mt = 0,
+  mb = 0,
+  ml = 0,
+  mr = 0,
+} = Astro.props;
+---
+
+<hr class={classMerge("gx-divider", className, `gx-divider-${orientation}`)} />
+
+<style lang="scss" define:vars={{ thickness, color, mt, mb, ml, mr }}>
+  .gx-divider {
+    border: 0;
+    margin: 0;
+    border-top: var(--thickness) solid var(--color);
+    margin-top: var(--mt);
+    margin-bottom: var(--mb);
+    margin-left: var(--ml);
+    margin-right: var(--mr);
```

Comment:
```suggestion
export interface DividerProps {
  class?: string;
  thickness?: string;
  orientation?: "horizontal" | "vertical";
  color?: string;
  mt?: OptionalSizeProp;
  mb?: OptionalSizeProp;
  ml?: OptionalSizeProp;
  mr?: OptionalSizeProp;
}

const {
  class: className,
  thickness = "1px",
  orientation = "horizontal",
  color = "#323232",
  mt,
  mb,
  ml,
  mr,
} = Astro.props;
---

<hr class={classMerge("gx-divider", className, `gx-divider-${orientation}`)} />

<style
  lang="scss"
  define:vars={{
    thickness,
    color,
    mt: getSizeProp(mt, 4),
    mb: getSizeProp(mb, 4),
    ml: getSizeProp(ml, 0),
    mr: getSizeProp(mr, 0),
  }}
>
  .gx-divider {
    border: 0;
    margin: 0;
    border-top: var(--thickness) solid var(--color);
    margin-top: var(--mt);
    margin-bottom: var(--mb);
    margin-left: var(--ml);
    margin-right: var(--mr);
```

The `getSizeProp` helper is implemented in https://github.com/gemini-ui/gemini-ui-monorepo/pull/16 - it would allow us to accept both exact values (e.g. "23.5px") as well as standard sizes (e.g. "4").